src/tests/system/tests/test_netgroups.py

```diff
@@ -108,3 +124,282 @@ def test_netgroups__add_remove_netgroup_member(client: Client, provider: Generic
+    :setup:
+        1. Create users, groups and start sssd.
+    :steps:
+        1. Check that (testhost5, ng9005, ldap.test) is present as a direct member of "DEVUsers".
+        2. Check that (testhost1, ng9000, ldap.test) is also present,
+            even though this tuple was added to "QAUsers", not "DEVUsers".
+            This confirms that the nested group membership is working correctly
+            (since "QAUsers" is nested within "DEVUsers").
+    :expectedresults:
+        1. (testhost5, ng9005, ldap.test) is present as a direct member of "DEVUsers".
+        2. (testhost1, ng9000, ldap.test) is present as a direct member of "DEVUsers".
+    """
+    ou = ldap.ou("Netgroup")
+    create_users(ldap)
+
+    qa_users = ldap.netgroup("QAUsers", basedn=ou).add()
+    qa_users.add_member(host="testhost1", user="ng9000", domain="ldap.test")
+
+    dev_users = ldap.netgroup("DEVUsers", basedn=ou).add()
+    dev_users.add_member(host="testhost5", user="ng9005", domain="ldap.test")
+    ldap.ldap.modify(dev_users.dn, add={"memberNisNetgroup": "QAUsers"})
+
+    client.sssd.dom("test")["entry_cache_timeout"] = "60"
+    client.sssd.start()
+
+    member = client.tools.getent.netgroup("DEVUsers").members
+    assert "(testhost5, ng9005, ldap.test)" in member
+    assert "(testhost1, ng9000, ldap.test)" in member
+
+
+@pytest.mark.importance("low")
+@pytest.mark.topology(KnownTopology.LDAP)
+def test_netgroup__add_dn_membernisnetgroup(client: Client, ldap: LDAP):
+    """
+    :title: Adding dn to memberNisNetgroup
+    :setup:
+        1. Create users, groups and start sssd.
+    :steps:
+        1. Check that the tuple (testhost5, ng9005, ldap.test) is present as a direct member of "DEVUsers".
+        2. Check that the tuple (testhost1, ng9000, ldap.test) is also present.
+            Since "QAUsers" is now referenced as part of "DEVUsers", its members
+            (like ng9000 on testhost1) are inherited by "DEVUsers".
+    :expectedresults:
+        1. Tuple (testhost5, ng9005, ldap.test) is present as a direct member of "DEVUsers".
+        2. Tuple (testhost1, ng9000, ldap.test) is also present.
+    :customerscenario: False
+    """
+    ou = ldap.ou("Netgroup")
+    create_users(ldap)
+
+    qa_users = ldap.netgroup("QAUsers", basedn=ou).add()
+    qa_users.add_member(host="testhost1", user="ng9000", domain="ldap.test")
+
+    dev_users = ldap.netgroup("DEVUsers", basedn=ou).add()
+    dev_users.add_member(host="testhost5", user="ng9005", domain="ldap.test")
+    ldap.ldap.modify(dev_users.dn, replace={"memberNisNetgroup": qa_users.dn})
+
+    client.sssd.dom("test")["entry_cache_timeout"] = "60"
+    client.sssd.start()
+
+    member = client.tools.getent.netgroup("DEVUsers").members
+    assert "(testhost5, ng9005, ldap.test)" in member
+    assert "(testhost1, ng9000, ldap.test)" in member
+
+
+@pytest.mark.importance("low")
+@pytest.mark.topology(KnownTopology.LDAP)
+def test_netgroup__different_syntax(client: Client, ldap: LDAP):
```

Comment:
All of these tests where you are checking syntax can be parameterized. Please look at https://github.com/SSSD/sssd/blob/b928dbe1fcb6cf4b7acbd97d3df9514d5c554953/src/tests/system/tests/test_failover.py#L19 as an example.